framework/src/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -515,17 +567,21 @@ public boolean applyUncheckedCodeDefaults(final Element annotationScope) {
             // to work instead of the
             // isElementFromByteCode/declarationFromElement/isFromStubFile calls,
             // but it doesn't work correctly and tests fail.
-            // (That whole @FromStubFile and @FromByteCode annotation
-            // logic should be replaced by something sensible.)
-            SourceChecker checker = atypeFactory.getContext().getChecker();
-            return (checker.hasOption("safeDefaultsForUnannotatedBytecode") &&
-                    ElementUtils.isElementFromByteCode(annotationScope) &&
-                    atypeFactory.declarationFromElement(annotationScope) == null &&
-                    !atypeFactory.isFromStubFile(annotationScope)) ||
-                   (checker.hasOption("useSafeDefaultsForUnannotatedSourceCode") &&
-                    !annotatedForThisChecker);
-        }
 
+            boolean isFromStubFile = atypeFactory.isFromStubFile(annotationScope);
+            boolean isBytecode = ElementUtils.isElementFromByteCode(annotationScope) &&
+                           atypeFactory.declarationFromElement(annotationScope) == null &&
+                           !isFromStubFile;
+            if (isBytecode) {
+                return useUncheckedCodeDefaultsBytecode;
+            } else if (isFromStubFile){
```

Comment:
Why are stub files treated separately? Shouldn't it just be handled like source code in the next branch?